src/file/tokio_compat_file.rs

```diff
@@ -463,12 +524,36 @@ impl AsyncWrite for TokioCompatFile {
         // sftp v3 cannot send more than self.max_write_len() data at once.
         let max_write_len = self.max_write_len_impl();
 
-        let n: u32 = buf
+        let mut n: u32 = buf
             .len()
             .try_into()
             .map(|n| min(n, max_write_len))
             .unwrap_or(max_write_len);
 
+        let write_limit = self.get_auxiliary().tokio_compat_file_write_limit();
+        let mut write_len = self.as_mut().write_len;
```

Comment:
I'm a bit confused why this works

`TokioCompatFile` does not implement `Unpin`, so `DerefMut` should not be implemented for `Pin<&mut TokioCompatFile>`

Reply:
It just gets the value of `write_len`, rather than reference. So it only requires `Deref` for `Pin<&mut TokiocompatFile>`.

Reply:
Thanks!
I missed that one.

Can you open another PR to clean up unnecessary use of project if ppss?

Reply:
Ok. I'd like to do it.